lib/checks/visibility/hidden-content.js

```diff
@@ -0,0 +1,17 @@
+if (!axe.commons.dom.isVisible(node)) {
```

Comment:
It seems to me this check is overly complicated. We don't have to check if the element is visible. You've already done that in the matcher. All this should be doing is check if there is content or not. If there is, return undefined, if not, return true.